src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6919,6 +6920,230 @@ def create_stack(self, stack: StackRequest) -> StackResponse:
+                                NotImplementedError,
+                                AuthorizationException,
+                            ) as e:
+                                raise RuntimeError(
+                                    f"Access to the resource could not be "
+                                    f"verified: {e}"
+                                )
+
+                            resources = connector_resources.resources[0]
+                            if resources.resource_ids:
+                                if len(resources.resource_ids) > 1:
+                                    raise RuntimeError(
+                                        f"Multiple {resource_type} resources are "
+                                        "available for the selected connector. "
+                                        "Please use the a specific resource-id to "
+                                        f"configure a {resource_type} resource."
+                                    )
+                                else:
+                                    resource_id = resources.resource_ids[0]
+
+                            component_update = ComponentUpdate(
+                                connector=service_connector.id,
+                                connector_resource_id=resource_id,
+                            )
+                            self.update_stack_component(
+                                component_id=component.id,
+                                component_update=component_update,
+                            )
+
+                    components_mapping[component_type] = [
+                        component.id,
+                    ]
+
+                # Stack
+                stack_name = full_stack.name
+                while True:
+                    try:
+                        stack_request = StackRequest(
+                            user=full_stack.user_id,
+                            workspace=full_stack.workspace_id,
+                            name=stack_name,
+                            description=full_stack.description,
+                            components=components_mapping,
+                        )
+                        stack_response = self.create_stack(stack_request)
+                        break
+                    except EntityExistsError:
+                        stack_name = (
+                            f"{full_stack.name}-{random_str(4)}".lower()
+                        )
+                        continue
+
+                handler.metadata.update(
+                    stack_response.get_analytics_metadata()
+                )
+            except Exception as e:
+                for service_connector_id in service_connectors_created_ids:
+                    self.delete_service_connector(
+                        service_connector_id=service_connector_id
+                    )
+                for component_id in components_created_ids:
```

Comment:
`components_created_ids` is declared inside the try-except block:

```
Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_stores/sql_zen_store.py", line 6965, in create_full_stack
    service_connector_response = self.create_service_connector(
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/zenml/analytics/utils.py", line 178, in inner_func
    result = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_stores/sql_zen_store.py", line 6102, in create_service_connector
    service_connector.validate_and_configure_resources(
  File "/opt/venv/lib/python3.11/site-packages/zenml/models/v2/core/service_connector.py", line 196, in validate_and_configure_resources
    _validate_and_configure_resources(
  File "/opt/venv/lib/python3.11/site-packages/zenml/models/v2/core/service_connector.py", line 1013, in _validate_and_configure_resources
    raise ValueError(
ValueError: connector configuration is not valid: missing required attribute 'role_arn'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_server/utils.py", line 447, in decorated
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_server/routers/workspaces_endpoints.py", line 417, in create_full_stack
    return zen_store().create_full_stack(full_stack)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.11/site-packages/zenml/zen_stores/sql_zen_store.py", line 7138, in create_full_stack
    for component_id in components_created_ids:
                        ^^^^^^^^^^^^^^^^^^^^^^
UnboundLocalError: cannot access local variable 'components_created_ids' where it is not associated with a value

```

Reply:
Good catch, the same issue applied to the connectors as well. Fixing them both now.